---
title: "Jenny Leopoldina Smith, MSc"
id: "Resume"
---

<br>

## <i class="bi bi-mortarboard"></i> Education

**Applied Bioinformatics and Genomics**, MSc

-   *University of Oregon*, Eugene, OR

**Secondary Education in Science**, MEd

-   *Arizona State University*, Phoenix, AZ

**Biology**, BA, *magna cum laude*

-   *University of San Diego*, San Diego, CA

## <i class="bi bi-briefcase"></i> Experience and Skills

::: panel-tabset
# <i class="bi bi-briefcase-fill"></i> Work Experience

**Computational Biologist, Senior**

Research Scientific Computing Team

*Seattle Children’s*, Seattle, WA

**April 2022 – Sept 2024**

-   Developed and adapted reproducible genomics workflows using Nextflow and nf-core tools for RNA-seq quantification, Cut&Run, ATAC-seq, and PacBio Isoseq;

-   Nextflow pipelines with containerized software with singularity/apptainer and the SLURM/PBSpro executors on a high-performance compute cluster (HPC).

-   Analysis and visualization of multi-sample single cell scRNA-seq and scATAC-seq datasets with Cellranger quantification, followed by doublet detection, ambient DNA correction, and dataset integration (SCANVI and Seurat v4/v5). Production of multi-omics analyses for bulk transcriptomic RNA-seq and single-cell datasets.

-   Generation of complex visualizations include heatmaps, circos plots, oncoprints, 3D scatter plots, genomic tracks, network graphs, and volcano plots, among others, to identify actionable insights from multivariate biological datasets from NGS and public databases (ensembl, UCSC, Genomic Data Commons).

-   Statistical analysis of complex clinical data elements using dplyr and tidyverse framework.

-   Engagement in bioinformatics support by providing guidance and troubleshooting assistance at office hours and teaching internal courses.

---

**Bioinformatics Analyst**

Principal Investigator: Soheil Meshinchi

*Fred Hutchinson Cancer Center*, Seattle, WA

**January 2017 – March 2022**

-   Development of genomics workflows for cloud computing on AWS using Nextflow for processing raw RNA-seq data for gene expression quantification and fusion detection.

-   Complex data manipulation and interpretation using dplyr and the principles of tidyverse on large datasets derived from NGS and public databases. Retrieval of genomics and clinical data from public databases using public APIs.

-   Profiled pediatric AML subtypes including gene fusions CBFA2T3-GLIS2, NUP98 fusions, and mutations (FLT3-ITD) using multi-omic data (RNA-sequencing and miRNA-seq) with supervised and unsupervised clustering ML algorithms, as well as statistical regression and classification.

-   Collaborated in a multidisciplinary environment with molecular biologists, bioinformaticians, and lead investigators. These bioinformatics analyses and data visualizations utilized in 18 manuscripts published in peer reviewed journals, as well as grant applications for American Cancer Society, Saint Baldricks Foundation, Gabriella Miller Foundation, and TpAML being funded.

-   Survival analysis of AML variants and application of the LSC17 prognostic score to a pediatric cohort using Kaplan-Meier estimates and Cox proportional hazards regression; resulted in oral presentation at American Society of Hematology (2017) and followed by collaborative analysis with UCSF.

---

**Postbaccalaureate Fellow**

Principal Investigator: Maria Morasso

*National Institutes of Health*, Bethesda, MD

**June 2014 - July 2016**

-   Conducted biomedical research on the function of homeodomain protein DLX3 at NIAMS.

-   Bioinformatics analysis of ATAC-seq and ChIP-seq from murine mouse models using command line tools (CLI) for genomic alignment, MACS2 peak calling, motif analysis with HOMER, and figure generation with NGS.plot and Deeptools.

---

**Science Teacher**

Teach for America Phoenix Corps

*Agua Fria High School*, Avondale, AZ

**June 2012 - May 2014**

-   Certified for secondary biology and chemistry education; Recognition of Excellence (ROE) Award for Biology Issued by Educational Testing Service (ETS).

-   Courses taught: general biology, AP biology, general chemistry and integrated science.

---

**Research Assistant**

Principal Investigator: Terry Bird

*University of San Diego*, San Diego, CA

**August 2009 - May 2012**

-   Investigated the role of CHPT in the cellular differentiation of the bacterium Rhodospirillum centenum

-   Utilized site-directed mutagenesis and construction of triple knockout mutants of CHPT, CTRA, and CYD2 using molecular cloning techniques.

# <i class="bi bi-pc-display"></i> Skills

### Experience with Software

-   R programming:

    -   base R, Tidyverse, Bioconductor

    -   R package development

    -   Quarto / Rmarkdown

    -   Posit workbench / Rstudio IDE

-   Bash shell scripting

-   Nextflow and nf-core

    -   Modularized DSL2

-   Python programming

    -   Scanpy and SCVI-tools

    -   Multi-lingual environments with Reticulate

-   Version control

    -   Git and github (gh) command line utilities

    -   Github and Bitbucket remotes

-   HPC schedulers

    -   PBS

    -   SLURM

-   Containerization

    -   Docker

    -   Apptainer / Singularity

-   Amazon Web Services: AWS S3 and Batch

### Bioinformatics and Statistics

-   Unsupervised clustering (dendrograms, PCA, NMDS, and UMAP)

-   Statistical regression, classification, regularization

-   Differential gene expression (Limma and EdgeR)

-   Survival and time-to-event analysis

-   Data visualization (ggplot2, ComplexHeatmap, plotly R, and Gviz)

### Experience with NGS Data

-   Transcriptomics

    -   scRNA-seq (single-cell RNA-seq)

    -   scATAC-seq (single-cell ATAC-seq)

    -   RNA-seq (Illumina short read, PacBio long read)

    -   miRNA-seq (microRNA-seq)

-   Epigenetics

    -   ATAC-seq

    -   Cut&Run

    -   ChIP-seq

# <i class="bi bi-backpack-fill"></i> Continuing Education and Volunteering

### Continuing Education and Volunteering

-   POSIT Conference, Seattle, WA, August 2024

-   Nextflow Hackathon, University of Washington, March 2024

-   International Society for Computational Biology (ISCB) member, 2023 - 2024

-   RLadies event organizer, Seattle, WA, 2018 - 2023

-   Bioconductor conference organizer, Seattle, WA, 2022

-   Mentor, UO Bioinformatics and Genomics Graduate Program, 2021 - 2022

-   SnpReportR package, Carnegie Mellon and DNAnexus Hackathon, January 2021

-   RLadies, Seattle, WA 2018-2021

-   Pacific Biosciences Isoseq Transcriptome Analysis Training, July 2020

-   Nextflow workflow development training, May 2020

-   Women In Biology, MAPS Mentorship Group, 2017-2019

-   ConsensusML, Machine learning classification in AML, NCBI Hackathon February 2018

-   Fred Hutch Summer High School Internship Mentor, June-August 2018

-   Summer Institute in Statistics for Big Data (SISBID), University of Washington, June 2017
:::
